pyomo/core/kernel/expr_coopr3.py

```diff
@@ -638,7 +638,11 @@ def to_string(self, ostream=None, verbose=None, precedence=0):
             ostream.write("( ")
         first = True
         if self._coef != 1:
+            if self._coef < 0:
+                ostream.write("(")
             ostream.write(str(self._coef))
+            if self._coef < 0:
+                ostream.write(")")
```

Comment:
Correct. For product expressions in pyomo, negative coefficients were simply written at the front of their expression, regardless of the coefficient related to this expression coming from the "higher up" expression (like a sum expression that contains a product expression, might write "blahblah - -5 * this * that == 0"). I made the change here since I figured this wasn't intended behavior, and this was the fix I thought of.